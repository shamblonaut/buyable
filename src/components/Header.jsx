import styled from "styled-components";

const Logo = styled.header`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 8px;
  font-family: Montserrat, system-ui, sans-serif;
`;

const LogoMark = styled.img`
  width: 48px;
  height: auto;
  padding: 0 8px;
`;

const Header = () => (
  <Logo>
    <LogoMark src="/logo.svg" alt="Buyable Logo" />
    <h1>Buyable</h1>
  </Logo>
);

export default Header;
