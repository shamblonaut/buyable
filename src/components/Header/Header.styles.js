import styled from "styled-components";

export const Brand = styled.header`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 8px;
  font-family: Montserrat, system-ui, sans-serif;
`;

export const Logo = styled.img`
  width: 48px;
  height: auto;
  padding: 0 8px;
`;

export const Title = styled.p`
  font-size: 2.5rem;
  font-weight: 700;
`;
