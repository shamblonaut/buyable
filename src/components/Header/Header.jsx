import { Brand, HomeLink, Logo, Title } from "./Header.styles";

const Header = () => (
  <Brand>
    <HomeLink to="/">
      <Logo src="/logo.svg" alt="Buyable Logo" />
      <Title>BUYABLE</Title>
    </HomeLink>
  </Brand>
);

export default Header;
