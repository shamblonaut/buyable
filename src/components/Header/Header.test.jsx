import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";

import { BrowserRouter } from "react-router-dom";

import Header from "./Header";

describe("Header component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
