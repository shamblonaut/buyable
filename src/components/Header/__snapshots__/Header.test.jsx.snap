// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders correctly 1`] = `
<div>
  <header
    class="sc-bRKDuR decPCC"
  >
    <a
      class="sc-hvigdm cvYFI"
      data-discover="true"
      href="/"
    >
      <img
        alt="Buyable Logo"
        class="sc-fhHczv dhZxPy"
        src="/logo.svg"
      />
      <p
        class="sc-ggWZvA bowmnC"
      >
        BUYABLE
      </p>
    </a>
  </header>
</div>
`;
